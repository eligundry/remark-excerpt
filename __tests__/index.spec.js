const {join} = require('path');

const {wrap} = require('jest-snapshot-serializer-raw');
const remark = require('remark');
const vfile = require('to-vfile');

const plugin = require('../index');

const processFixture = async (name, options) => {
    const path = join(__dirname, 'fixtures', name);
    const file = await vfile.read(path);
    const result = await remark()
        .use(plugin, options)
        .process(file);

    return result.toString();
};

test('should return unmodified document when no excerpt exists', async () => {
    const result = await processFixture('no-excerpt.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "exceprt" keyword', async () => {
    const result = await processFixture('excerpt.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "more" keyword', async () => {
    const result = await processFixture('more.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "preview" keyword', async () => {
    const result = await processFixture('preview.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using custom keyword', async () => {
    const options = {
        keyword: 'custom'
    };
    const result = await processFixture('custom.md', options);

    expect(wrap(result)).toMatchSnapshot();
});

test('should return after excerpt when multiple exist', async () => {
    const result = await processFixture('multiple.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should handle no spacing in exceprt comment', async () => {
    const result = await processFixture('no-space-in-comment.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should handle mdx', async () => {
    const result = await processFixture('mdx.mdx');

    expect(wrap(result)).toMatchSnapshot();
});
