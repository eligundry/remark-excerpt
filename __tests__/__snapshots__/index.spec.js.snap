// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle keyword with dashes 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should handle keyword with spaces 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should handle mdx 1`] = `
import something from 'something';

export const meta = {
    prop: 'value'
};

# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should handle no spacing in exceprt comment 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return after excerpt when multiple exist 1`] = `
# Title

Paragraph 1.

`;

exports[`should return excerpt using "exceprt" keyword 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using "more" keyword 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using "preview" keyword 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using custom keyword 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return unmodified document when no excerpt exists 1`] = `
# Title

Paragraph 1.

Paragraph 2.

Paragraph 3.

Paragraph 4.

`;
